### PR TITLE
[SYCL] E2E test to retain PI events until they have signaled

### PR DESCRIPTION
**Pull request by smaslov-intel**
Covers the change in https://github.com/intel/llvm/pull/3350
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: SYCL/Plugin/retain_events.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    constexpr size_t N = 1024 * 1000;
+
+    int * src = sycl::malloc_shared<int>(N, D, C);
+    int * dst = sycl::malloc_shared<int>(N, D, C);
+
+    std::iota(src, src + N, 0);
+    std::fill(dst, dst + N, 0);
+
+    for (int i = 0; i < 256; ++i) {
+        sycl::queue Q { C , D, async_sycl_error };//, sycl::property::queue::enable_profiling() };
```
Comment (bso-intel) [inline SYCL/Plugin/retain_events.cpp]:
```suggestion
        sycl::queue Q { C , D, async_sycl_error };
```

**Review by bso-intel — DISMISSED** [review]
LGTM

**Comment by vladimirlaz** [comment]
@smaslov-intel, could you please fix clang-format issue?

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
@smaslov-intel, could you please fix clang-format issue?

**Review by vladimirlaz — APPROVED** [review]

File: SYCL/Plugin/retain_events.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    Q.submit([&](sycl::handler &SH) {
+      sycl::range<1> r(N);
+      SH.parallel_for<class X>(r, [=](sycl::id<1> id) {
+        size_t i = id.get(0);
+        dst[i] = src[i] + 1;
+      });
+    });
+  }
+
```
Comment (bader) [inline SYCL/Plugin/retain_events.cpp:61]: @smaslov-intel, @againull, shouldn't we set SYCL_PI_LEVEL_ZERO_TRACK_INDIRECT_ACCESS_MEMORY or wait for all kernels to complete before freeing USM memory to make this test valid?

Comment (bader) [reply SYCL/Plugin/retain_events.cpp:61]: @alexbatashev FYI.

**Pull request merged**